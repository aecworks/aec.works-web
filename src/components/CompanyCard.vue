<template>
  <!-- <Card @click="handleClick(company)"> -->
  <Card>
    <template v-slot:logo>
      <img v-if="company.logoUrl" class="company-logo" :src="getImageUrl(company.logoUrl)" />
    </template>
    <template v-slot:cover>
      <img v-if="company.coverUrl" :src="getImageUrl(company.coverUrl)" />
    </template>
    <template v-slot:default>
      <!-- Company Name -->
      <h2 class="mt-1">
        <!-- TODO throwing router error -->
        <router-link :to="{ name: 'Company', params: { slug: company.slug || '?' } }">
          {{ company.name }}
        </router-link>
      </h2>

      <!-- Company Description -->
      <p class="mt-1">{{ company.description || '-' }}</p>

      <!-- Company Hashtags -->
      <div class="mt-1">
        <Hashtag
          v-for="slug in company.hashtags"
          :key="slug"
          :slug="slug"
          clickable
          @click="$emit('hashtagClick', slug)"
        />
      </div>

      <!-- Footer -->
      <div class="flex mt-1">
        <Icon :icon="'chat'" class="mr-1">
          <span class="small">{{ company.threadSize }}</span>
        </Icon>
        <Icon :icon="'clap'" clickable @click="handleClapClick(company)">
          <span class="small">{{ localClapCount || company.clapCount }}</span>
        </Icon>
        <span class="small flex-right">{{ company.location }}</span>
      </div>
    </template>
  </Card>
</template>

<script>
import api from '@/api'
import { waitForLogin } from '@/mixins'
import Icon from '@/components/Icon.vue'
import Card from '@/components/Card.vue'
import Hashtag from '@/components/Hashtag'

export default {
  name: 'CompanyCard',
  components: { Hashtag, Card, Icon },
  props: ['company'],
  data() {
    return {
      localClapCount: null,
    }
  },
  methods: {
    handleClick(company) {
      this.$router.push({ name: 'Company', params: { slug: company.slug } })
    },
    getImageUrl(logo) {
      if (logo) {
        return logo
      } else {
        return 'https://api.adorable.io/avatars/285/abott@adorable.png'
      }
    },
    async handleClapClick(company) {
      await waitForLogin()
      const clapCount = await api.postCompanyClap(company.slug)
      this.localClapCount = clapCount
    },
  },
}
</script>

<style lang="scss" scoped>
.company-logo {
  position: absolute;
  right: 50px;
  bottom: -50px;
  width: 80px;
  height: 80px;
  @extend .border-thin;
  // Custom:
  border-radius: 5px;
  border-color: $dark-gray;
  border-width: 1px;
  box-shadow: 0 5px 10px -10px rgba(0, 0, 0, 0.25);
  background-color: white;
}
</style>
