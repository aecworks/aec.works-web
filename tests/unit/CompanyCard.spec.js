import { shallowMount, RouterLinkStub } from '@vue/test-utils'
import CompanyCard from '@/components/CompanyCard.vue'

const FULL_COMPANY = {
  slug: '3d-repo',
  createdBy: {
    slug: 'testuser',
    name: 'Joseph Banks',
    bio:
      'Politics ball little miss admit maintain couple.\nInclude parent draw challenge total. Training avoid later design tough. Left stuff family pay health because every.',
    location: 'New Dianatown',
    avatarUrl: null,
  },
  threadId: 11,
  createdAt: '2020-09-14T05:23:32.205711Z',
  clapCount: 1,
  threadSize: 2,
  lastRevisionId: 1,
  articles: [],
  logoUrl:
    'https://aecworks-bucket-staging.s3.amazonaws.com/__sized__/images/29930a4516ed439fb07ec38413464efd-crop-c0-5__0-5-80x80.png',
  coverUrl:
    'https://aecworks-bucket-staging.s3.amazonaws.com/__sized__/images/fede69a591e84008b535767e2212632c-crop-c0-5__0-5-800x400.png',
  name: '3D Repo',
  description: 'Cloud-Based BIM',
  website: 'https://3drepo.com',
  location: 'London, UK',
  twitter: null,
  crunchbaseId: '3d-repo',
  logo: 530,
  cover: 531,
  hashtags: ['webapp', 'issuetracking', 'changemanagement'],
}

const factory = (company = FULL_COMPANY) =>
  shallowMount(CompanyCard, {
    propsData: {
      company,
    },
    stubs: {
      NuxtLink: RouterLinkStub,
    },
  })

describe('CompanyCard.vue', () => {
  it('Renders a full company with logo and cover', () => {
    const wrapper = factory(FULL_COMPANY)
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('Renders a card without a cover', () => {
    const company = { ...FULL_COMPANY, coverUrl: null }
    const wrapper = factory(company)
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('Renders a card without a logo', () => {
    const company = { ...FULL_COMPANY, logoUrl: null }
    const wrapper = factory(company)
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('Renders a card without a cover or a logo', () => {
    const company = { ...FULL_COMPANY, logoUrl: null, coverUrl: null }
    const wrapper = factory(company)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
