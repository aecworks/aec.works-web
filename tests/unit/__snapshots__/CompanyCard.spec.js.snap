// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompanyCard.vue Renders a card without a cover 1`] = `
<card-stub banner="" showimage="true" data-cy="company-card">
  <lazyimg-stub src="https://aecworks-bucket-staging.s3.amazonaws.com/__sized__/images/29930a4516ed439fb07ec38413464efd-crop-c0-5__0-5-80x80.png" class="company-logo"></lazyimg-stub>
  <!---->
  <h2 class="mt-1">
    <router-link-stub to="[object Object]" tag="a" ariacurrentvalue="page" event="click">
      3D Repo
    </router-link-stub>
  </h2>
  <p class="mt-1">Cloud-Based BIM</p>
  <div class="mt-1">
    <hashtag-stub slug="webapp" clickable="true"></hashtag-stub>
    <hashtag-stub slug="issuetracking" clickable="true"></hashtag-stub>
    <hashtag-stub slug="changemanagement" clickable="true"></hashtag-stub>
  </div>
  <div class="flex mt-1">
    <icon-stub icon="chat" class="mr-1"><span class="small">2</span></icon-stub>
    <icon-stub clickable="true" icon="clap"><span class="small">1</span></icon-stub> <span class="small flex-right">London, UK</span>
  </div>
</card-stub>
`;

exports[`CompanyCard.vue Renders a card without a cover or a logo 1`] = `
<card-stub banner="" showimage="true" data-cy="company-card">
  <!---->
  <!---->
  <h2 class="mt-1">
    <router-link-stub to="[object Object]" tag="a" ariacurrentvalue="page" event="click">
      3D Repo
    </router-link-stub>
  </h2>
  <p class="mt-1">Cloud-Based BIM</p>
  <div class="mt-1">
    <hashtag-stub slug="webapp" clickable="true"></hashtag-stub>
    <hashtag-stub slug="issuetracking" clickable="true"></hashtag-stub>
    <hashtag-stub slug="changemanagement" clickable="true"></hashtag-stub>
  </div>
  <div class="flex mt-1">
    <icon-stub icon="chat" class="mr-1"><span class="small">2</span></icon-stub>
    <icon-stub clickable="true" icon="clap"><span class="small">1</span></icon-stub> <span class="small flex-right">London, UK</span>
  </div>
</card-stub>
`;

exports[`CompanyCard.vue Renders a card without a logo 1`] = `
<card-stub banner="" showimage="true" data-cy="company-card">
  <!---->
  <lazyimg-stub src="https://aecworks-bucket-staging.s3.amazonaws.com/__sized__/images/fede69a591e84008b535767e2212632c-crop-c0-5__0-5-800x400.png"></lazyimg-stub>
  <h2 class="mt-1">
    <router-link-stub to="[object Object]" tag="a" ariacurrentvalue="page" event="click">
      3D Repo
    </router-link-stub>
  </h2>
  <p class="mt-1">Cloud-Based BIM</p>
  <div class="mt-1">
    <hashtag-stub slug="webapp" clickable="true"></hashtag-stub>
    <hashtag-stub slug="issuetracking" clickable="true"></hashtag-stub>
    <hashtag-stub slug="changemanagement" clickable="true"></hashtag-stub>
  </div>
  <div class="flex mt-1">
    <icon-stub icon="chat" class="mr-1"><span class="small">2</span></icon-stub>
    <icon-stub clickable="true" icon="clap"><span class="small">1</span></icon-stub> <span class="small flex-right">London, UK</span>
  </div>
</card-stub>
`;

exports[`CompanyCard.vue Renders a full company with logo and cover 1`] = `
<card-stub banner="" showimage="true" data-cy="company-card">
  <lazyimg-stub src="https://aecworks-bucket-staging.s3.amazonaws.com/__sized__/images/29930a4516ed439fb07ec38413464efd-crop-c0-5__0-5-80x80.png" class="company-logo"></lazyimg-stub>
  <lazyimg-stub src="https://aecworks-bucket-staging.s3.amazonaws.com/__sized__/images/fede69a591e84008b535767e2212632c-crop-c0-5__0-5-800x400.png"></lazyimg-stub>
  <h2 class="mt-1">
    <router-link-stub to="[object Object]" tag="a" ariacurrentvalue="page" event="click">
      3D Repo
    </router-link-stub>
  </h2>
  <p class="mt-1">Cloud-Based BIM</p>
  <div class="mt-1">
    <hashtag-stub slug="webapp" clickable="true"></hashtag-stub>
    <hashtag-stub slug="issuetracking" clickable="true"></hashtag-stub>
    <hashtag-stub slug="changemanagement" clickable="true"></hashtag-stub>
  </div>
  <div class="flex mt-1">
    <icon-stub icon="chat" class="mr-1"><span class="small">2</span></icon-stub>
    <icon-stub clickable="true" icon="clap"><span class="small">1</span></icon-stub> <span class="small flex-right">London, UK</span>
  </div>
</card-stub>
`;
